messages/teleporter/message_handler.go

```diff
@@ -183,47 +180,36 @@ func (m *messageHandler) ShouldSendMessage(destinationClient vms.DestinationClie
 		return false, nil
 	}
 
-	if decision, err := m.getShouldSendMessageFromDecider(); err != nil {
+	decision, err := m.getShouldSendMessageFromDecider()
+	if err != nil {
 		m.logger.Warn(
 			"Error delegating to decider",
 			zap.String("warpMessageID", m.unsignedMessage.ID().String()),
 			zap.String("teleporterMessageID", teleporterMessageID.String()),
 		)
-	} else {
-		if !decision {
-			m.logger.Info(
-				"Decider rejected message",
-				zap.String("warpMessageID", m.unsignedMessage.ID().String()),
-				zap.String("teleporterMessageID", teleporterMessageID.String()),
-				zap.String("destinationBlockchainID", destinationBlockchainID.String()),
-			)
-		}
-		return decision, nil
+		return true, nil
 	}
-
-	return true, nil
+	if !decision {
+		m.logger.Info(
+			"Decider rejected message",
+			zap.String("warpMessageID", m.unsignedMessage.ID().String()),
+			zap.String("teleporterMessageID", teleporterMessageID.String()),
+			zap.String("destinationBlockchainID", destinationBlockchainID.String()),
+		)
+	}
+	return decision, nil
 }
 
-// queries the decider service to determine whether this message should be
-// sent. if the decider client is nil, returns true.
+// Queries the decider service to determine whether this message should be
+// sent. If the decider client is nil, returns true.
 func (m *messageHandler) getShouldSendMessageFromDecider() (bool, error) {
 	deciderClientValue := reflect.ValueOf(m.deciderClient)
 
 	if !deciderClientValue.IsValid() || deciderClientValue.IsNil() {
 		return true, nil
 	}
```

Comment:
Since we're no longer directly assigning nil to `deciderClient`, we should be able to ditch reflection here: https://stackoverflow.com/a/13476765
```suggestion
	if m.deciderClient == nil {
		return true, nil
	}
```

Reply:
I implemented [Geoff's suggestion to have an "empty client" implementation](https://github.com/ava-labs/awm-relayer/pull/344#discussion_r1692063683), so now the interface is always valid, so there's no need to do any check in this method. see https://github.com/ava-labs/awm-relayer/pull/344/commits/f8662fdaf33fa9d196655df3ef485144d089e69d